waspc/cli/src/Wasp/Cli/Command/Start.hs

```diff
@@ -26,12 +26,12 @@ start = do
   InWaspProject waspRoot <- require
   let outDir = waspRoot </> Common.dotWaspDirInWaspProjectDir </> Common.generatedCodeDirInDotWaspDir
 
-  DbConnectionEstablished <- require
-
   cliSendMessageC $ Msg.Start "Starting compilation and setup phase. Hold tight..."
 
   warnings <- compile
 
+  DbConnectionEstablished <- require
```

Comment:
Ok, good, this should indeed happen after compilation.
Would be interesting to consider if this is something we could enforce at the type level. I am sure we could, but I guess the question is how pratical it would be.